File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/inbound/web/mission/dto/UserMissionResponse.java
package com.study.webflux.rag.infrastructure.inbound.web.mission.dto;

import java.time.Instant;

import com.study.webflux.rag.domain.mission.model.UserMission;

public record UserMissionResponse(
	String userId,
	String missionId,
	String status,
	Instant completedAt,
	Instant rewardedAt
) {
	public static UserMissionResponse from(UserMission userMission) {
		return new UserMissionResponse(
			userMission.userId().value(),
			userMission.missionId().value(),
			userMission.status().name(),
			userMission.completedAt(),
			userMission.rewardedAt());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/mission/dto/UserMissionResponse.kt
package com.miyou.app.infrastructure.inbound.web.mission.dto

import com.miyou.app.domain.mission.model.UserMission
import java.time.Instant

data class UserMissionResponse(
    val userId: String,
    val missionId: String,
    val status: String,
    val completedAt: Instant?,
    val rewardedAt: Instant?,
) {
    companion object {
        fun from(userMission: UserMission): UserMissionResponse =
            UserMissionResponse(
                userMission.userId.value,
                userMission.missionId.value,
                userMission.status.name,
                userMission.completedAt,
                userMission.rewardedAt,
            )
    }
}
